src/Middleware.php

```diff
@@ -72,7 +72,7 @@ function (ResponseInterface $response) use ($request) {
     /**
      * Middleware that pushes history data to an ArrayAccess container.
      *
-     * @param array|\ArrayAccess $container Container to hold the history (by reference).
+     * @param null|array|\ArrayAccess<int, array> $container Container to hold the history (by reference).
```

Comment:
Why is this allowed null?

Reply:
You are right, `null` is not a allowed value as that will cause an `InvalidArgumentException`.

Fixed in 969a11de1411491f0a02c5f1e722fa0d0fce7736

Reply:
but why did we move the docblock inline below?

Reply:
This change fixes a PHPStan error stating that `!\is_array($container) && !$container instanceof \ArrayAccess` can never be `false`.

As there is no typehint, the parameter's value *can* be null. The reason for putting the docblock inline is to not leak that information to outside of the named `history` constructor's method signature.

Let me know how to otherwise address this 👍

Reply:
> As there is no typehint, the parameter's value can be null.

Well it can be anything else as well.. not only null.. I believe leaving the phpstan error back is fine..